### PR TITLE
Pull events from v1/api/events

### PR DESCRIPTION
**Pull request by krx252525**

### CODE REVIEW

File: sumologic-k8s-api-cronjob-rbac.yaml [hunk 1]
```diff
@@ -17,12 +17,17 @@ spec:
           restartPolicy: OnFailure
           containers:
           - name:  sumologic-k8s-api
-            image: frankreno/sumologic-k8s-api:latest
+            imagePullPolicy: Always
+            image: mcluskey/sumologic-k8s-api:events
```
Comment (frankreno) [inline sumologic-k8s-api-cronjob-rbac.yaml]: can we revert these to previous values? Once we get the script changes in i will update docker hub image.

File: sumologic-k8s-api-cronjob-rbac.yaml [hunk 2]
```diff
@@ -17,12 +17,17 @@ spec:
...
             env:
             - name: SUMO_HTTP_URL
               value: <INSERT_URL_HERE>
             - name: K8S_API_URL
               value: http://127.0.0.1:8001
+            - name: X-Sumo-Category
+              value: k8s/api
+            - name: X-Sumo-Name
+              value: sumologic-k8s-api
```
Comment (frankreno) [inline sumologic-k8s-api-cronjob-rbac.yaml]: let's also remove these and not provide defaults.

File: sumologic-k8s-api-cronjob.yaml [hunk 3]
```diff
@@ -16,13 +16,18 @@ spec:
           restartPolicy: OnFailure
           containers:
           - name:  sumologic-k8s-api
-            image: frankreno/sumologic-k8s-api:latest
+            imagePullPolicy: Always
+            image: frankreno/sumologic-k8s-api:events
```
Comment (frankreno) [inline sumologic-k8s-api-cronjob.yaml]: same as above.

File: sumologic-k8s-api-cronjob.yaml [hunk 4]
```diff
@@ -16,13 +16,18 @@ spec:
...
             env:
             - name: SUMO_HTTP_URL
               value: <INSERT_URL_HERE>
             - name: K8S_API_URL
               value: http://127.0.0.1:8001
+            - name: X-Sumo-Category
+              value: k8s/api
```
Comment (frankreno) [inline sumologic-k8s-api-cronjob.yaml]: same as above.

**Review by frankreno — CHANGES_REQUESTED** [review]
Happy to add this, just some tweaks needed to the yamls..

File: sumologic-k8s-api-cronjob-rbac.yaml (hunk 1 above)
Comment (krx252525) [reply sumologic-k8s-api-cronjob-rbac.yaml]: Whoops, this was an accident... I wasn't really thinking this would get merged.

**Review by krx252525 — COMMENTED** [review]
Submitting requested changes

**Comment by krx252525** [comment]
@frankreno my only concern with this being the source of events for driving the dashboards/panels is that events have a [1-hour TTL](https://github.com/kubernetes/kubernetes/blob/5f5d4fd450ea086001f9f8a7b77605df0faff6d2/cmd/kube-apiserver/app/options/options.go#L94) meaning that we'll report a single event ~12 times if the cron is set to `*/5 * * * *`.

The [Datadog agent](https://github.com/DataDog/dd-agent/tree/877d72203920e6befea1d4adf231cd6058e1d2af/utils/kubernetes) uses leader election and a DaemonSet to provide resiliency and has some logic in order to provide deduplication. Leveraging similar logic seems like a better solution to provide to the Sumo community. Thoughts?

**Comment by frankreno** [comment]
I do agree. It is on my list to capture events, but due to the duplication this would create, I suggest we close this issue until we can solve properly.  If this is something you need, another option would be to make it configurable in the script via a flag that way it does not send unless you explicitly tell us to and execept the problems with this implementation

**Comment by krx252525** [comment]
I'm of the opinion that this shouldn't be merged since it causes too much duplication with data storage  and ingress being a premium in Sumologic I don't want someone to run perhaps without knowing the consequences. I'll close it and report back here or in SumoLogic/fluentd-kubernetes-sumologic with a proper solution.

Cheers @frankreno

**Pull request closed without merging**